server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -235,13 +238,23 @@ public RemoteClusterStateManifestInfo writeFullMetadata(ClusterState clusterStat
             isPublicationEnabled,
             isPublicationEnabled ? clusterState.customs() : Collections.emptyMap(),
             isPublicationEnabled,
-            remoteRoutingTableService.getIndicesRouting(clusterState.getRoutingTable())
+            remoteRoutingTableService.getIndicesRouting(clusterState.getRoutingTable()),
+            null
+        );
+
+        ClusterStateDiffManifest clusterStateDiffManifest = new ClusterStateDiffManifest(
+            clusterState,
+            ClusterState.EMPTY_STATE,
+            null,
+            uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata != null
+                ? uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata.getUploadedFilename()
+                : null
```

Comment:
If we are sending diff as null here, then will we have filename at all?

Reply:
Not that is redundant check, removed it.